src/memex/search/index.py

```diff
@@ -146,43 +150,35 @@ def _prepare(self, annotation):
 
         return (action, data)
 
-    def _stream_all_annotations(self, chunksize=2000):
-        # This is using a windowed query for loading all annotations in batches.
-        # It is the most performant way of loading a big set of records from
-        # the database while still supporting eagerloading of associated
-        # document data.
-
-        updated = self._base_query(). \
-                options(load_only('updated')). \
-                execution_options(stream_results=True). \
-                filter_by(deleted=False). \
-                order_by(models.Annotation.updated.desc()).all()
-
-        count = len(updated)
-        windows = [Window(start=updated[min(x+chunksize, count)-1].updated,
-                          end=updated[x].updated)
-                   for x in xrange(0, count, chunksize)]
-        basequery = self._eager_loaded_query(). \
-            order_by(models.Annotation.updated.asc())
-
-        for window in windows:
-            in_window = models.Annotation.updated.between(window.start, window.end)
-            for a in basequery.filter(in_window):
-                yield a
-
-    def _stream_filtered_annotations(self, annotation_ids):
-        annotations = self._eager_loaded_query(). \
-            execution_options(stream_results=True). \
-            filter(models.Annotation.id.in_(annotation_ids))
-
-        for a in annotations:
+
+def _all_annotations(session, windowsize=2000):
+    # This is using a windowed query for loading all annotations in batches.
+    # It is the most performant way of loading a big set of records from
+    # the database while still supporting eagerloading of associated
+    # document data.
+    not_deleted = sa.not_(models.Annotation.deleted)
+    windows = column_windows(session=session,
+                             column=models.Annotation.updated,  # implicit ASC
+                             windowsize=windowsize,
+                             where=not_deleted)
+    query = _eager_loaded_annotations(session).filter(not_deleted)
+
+    for window in windows:
+        for a in query.filter(window):
             yield a
 
-    def _base_query(self):
-        return self.session.query(models.Annotation).filter_by(deleted=False)
 
-    def _eager_loaded_query(self):
-        return self._base_query().options(
-            subqueryload(models.Annotation.document).subqueryload(models.Document.document_uris),
-            subqueryload(models.Annotation.document).subqueryload(models.Document.meta)
-        )
+def _filtered_annotations(session, ids):
+    annotations = (_eager_loaded_annotations(session)
+                   .execution_options(stream_results=True)
+                   .filter(models.Annotation.id.in_(ids)))
```

Comment:
I think this needs to filter out deleted annotations as well, since the batch indexer can be called with a list of annotation ids. We should prevent that from indexing deleted annotations.